ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/ConversionsSpec.scala

```diff
@@ -104,123 +108,227 @@ class ConversionsSpec extends AnyWordSpec with Matchers with OptionValues {
             ),
             (
               Rejection.CausalMonotonicityViolated,
               Code.ABORTED,
+              Map.empty,
             ),
             (
               Rejection.SubmittingPartyNotKnownOnLedger(Ref.Party.assertFromString("party")),
               Code.INVALID_ARGUMENT,
+              Map.empty,
             ),
             (
               Rejection.PartiesNotKnownOnLedger(Seq.empty),
               Code.INVALID_ARGUMENT,
+              Map.empty,
             ),
             (
-              Rejection.SubmitterCannotActViaParticipant(
-                Ref.Party.assertFromString("party"),
-                Ref.ParticipantId.assertFromString("id"),
+              Rejection.MissingInputState(partyStateKey("party")),
+              Code.ABORTED,
+              Map("key" -> "party: \"party\"\n"),
+            ),
+            (
+              Rejection.RecordTimeOutOfRange(Instant.EPOCH, Instant.EPOCH),
+              Code.ABORTED,
+              Map(
+                "minimum_record_time" -> Instant.EPOCH.toString,
+                "maximum_record_time" -> Instant.EPOCH.toString,
               ),
-              Code.PERMISSION_DENIED,
+            ),
+            (
+              Rejection.SubmittingPartyNotKnownOnLedger(party0),
+              Code.INVALID_ARGUMENT,
+              Map("submitter_party" -> party0),
+            ),
+            (
+              Rejection.PartiesNotKnownOnLedger(Iterable(party0, party1)),
+              Code.INVALID_ARGUMENT,
+              Map("parties" -> s"""[\"$party0\",\"$party1\"]"""),
             ),
           )
-        ) { (rejection, expectedCode) =>
-          {
-            val encodedEntry = Conversions
-              .encodeTransactionRejectionEntry(
-                submitterInfo,
-                rejection,
-              )
-              .build()
-            val finalReason = Conversions
-              .decodeTransactionRejectionEntry(encodedEntry)
-              .value
-            finalReason.code shouldBe expectedCode.value()
-            finalReason.definiteAnswer shouldBe false
-          }
+        ) { (rejection, expectedCode, expectedAdditionalDetails) =>
+          val encodedEntry = Conversions
+            .encodeTransactionRejectionEntry(
+              submitterInfo,
+              rejection,
+            )
+            .build()
+          val finalReason = Conversions
+            .decodeTransactionRejectionEntry(encodedEntry)
+            .value
+          finalReason.code shouldBe expectedCode.value()
+          finalReason.definiteAnswer shouldBe false
+          val actualDetails = finalReasonToDetails(finalReason)
+          actualDetails should contain allElementsOf (expectedAdditionalDetails)
+        }
+      }
+
+      "produce metadata that can be easily parsed" in {
+        forAll(
+          Table[Rejection, String, String => Any, Any](
+            ("rejection", "metadataKey", "metadataParser", "expectedParsedMetadata"),
```

Comment:
Nit: for consistency, you might want to use English expressions instead of camelCasedOnes here as well (e.g., `metadataKey` => `metadata key`).